### ISSUE TITLE
[HOLD for payment 2023-12-04] [$1000] Web - Displays "Hmm... it's not here"  when click on Task by Anonymous User in a public room

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#23491]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Create a task in any public room
2. Send any message in the chat report of task details
3. Back to the public room and copy public room URL
4. Open the public room URL in incognito mode
5. Tap on Task to go to Task details page

## Expected Result:
Since tasks are essentially a thread with a done state - the anonymous user should be able to view the Task details, but should not be able to take any of the actions related to a task (complete task, edit)

## Actual Result:
App displays "Hmm... it's not here"

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.44.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/757a69a0-3da5-4f0d-90ad-89e33abc821d

**Expensify/Expensify Issue URL:**
**Issue reported by:** @hoangzinh
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1689950784101419

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0184f834df2c94f428</li>
        <li>Upwork Job ID: 1684551229979066368</li>
        <li>Last Price Increase: 2023-08-03</li>
    </ul>
</details>

**Comment by StevenKKC** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Displays "Hmm... it's not here" when click on Task by Anonymous User in a public room.

### What is the root cause of that problem?
When click on task, the following code is called.
https://github.com/Expensify/App/blob/ff54c9e3775c3c9bd7f08baaba1c7cbb8e715467/src/components/ReportActionItem/TaskPreview.js#L75-L76
There is no checking whether is anonymous user.

### What changes do you think we should make in order to solve the problem?
We should add session check as below.
```js
	onPress={Session.checkIfActionIsAllowed(() => Navigation.navigate(ROUTES.getReportRoute(props.taskReportID)))}
```
### What alternative solutions did you explore? (Optional)
None.

**Comment by thienlnam** [comment]
Since tasks are essentially a thread with a done state - the anonymous user should be able to view the thread details, but should not be able to take any of the actions related to a task (complete task, edit)
So we should get a bug created to fix this so that an anonymous user can view task details in a public room always as the expected result

**Comment by michaelhaxhiu** [comment]
Agree 100% with @thienlnam

**Comment by BhuvaneshPatil** [comment]
This needs to be worked upon the backend.
currently we don't have permission to fetch the report.
<img width="507" alt="Screenshot 2023-07-27 at 11 00 28 PM" src="https://github.com/Expensify/App/assets/27822551/bedd53a9-fdd8-47c0-a89e-2b9cbfa83bba">

**Comment by abdulrahuman5196** [comment]
Reviewing today

**Comment by abdulrahuman5196** [comment]
@thienlnam I am too seeing this error on openReport api when click on task from public room by anonymous user. We should check on this in backend.

On a side note: During testing for the first time the task details actually opened for me. But after that I am seeing this issue. Not sure if backend sent something different during first time. 

**So we should check in backend first**

**Comment by thienlnam** [comment]
Yeah agreed, let's get this internal and see what the issue is

**Comment by michaelhaxhiu** [comment]
Going `internal` per the guidance above 🙏

**Comment by michaelhaxhiu** [comment]
Note: I'm preparing to go OOO for ~2 weeks and need a BZ buddy to watch over this in the meantime.

Next step: Please ensure that we push this forward from an `Internal` perspective in the next week to avoid tripping WAQ.

Thanks in advance @sophiepintoraetz, I can take this back when I get back if it's not complete before then.

**Comment by marcochavezf** [comment]
Hmm we should hide the composer input for anonymous users, looks like a regression

**Comment by sophiepintoraetz** [comment]
@marcochavezf - if it's a regression, can we hunt for the PR? cc @abdulrahuman5196 as well 🙏

**Comment by thienlnam** [comment]
> Hmm we should hide the composer input for anonymous users, looks like a regression

This is unrelated to the composer, it's that anonymous users are unable to open the task when it's created in a public room

> On a side note: During testing for the first time the task details actually opened for me. But after that I am seeing this issue. Not sure if backend sent something different during first time.


It looks like sometimes it works, but is inconsistent for some reason

**Comment by BhuvaneshPatil** [comment]
From my observation, if we open the task report that doesn't have any other comment (just fresh task),
But if we perform some action. Like mark as done ir add a reply for that thread. It shows the error.
Please confirm this behavior atbyour end as well.

**Comment by thienlnam** [comment]
Anonymous users shouldn't be able to take any actions on a task besides opening it to read it. They should actually be redirected to the login page, but I believe that is getting addressed in a different issue

**Comment by BhuvaneshPatil** [comment]
I think you misunderstood the Comment.

The task on which action performed (not in anonymous view), but by creator or anyone who has access

Then that task will give error when opening in anonymous.

**Comment by thienlnam** [comment]
Oh yes, that makes more sense - thanks for the context

**Comment by sophiepintoraetz** [comment]
@marcochavezf or @thienlnam - can you confirm that this is an internal fix/am I assigning you, Jack?

**Comment by thienlnam** [comment]
We still need to have someone investigate what the issue is internally - @marcochavezf is currently assigned

**Comment by marcochavezf** [comment]
> This is unrelated to the composer, it's that anonymous users are unable to open the task when it's created in a public room

Oh I jumped quickly to a bad conclusion by just reading the title and watching the first part of the video, apologies.

>  We still need to have someone investigate what the issue is internally - @marcochavezf is currently assigned

Yeah, I will check it out. I have a few other higher-priority issues, so I think I could investigate it next week

**Comment by sophiepintoraetz** [comment]
Just waiting on Marco to get to this!

**Comment by marcochavezf** [comment]
No update, focused on wave7

**Comment by marcochavezf** [comment]
No update, focused on bugs with customers and wave7, moving this to weekly meanwhile

**Comment by abdulrahuman5196** [comment]
@marcochavezf Can we move this to weekly again, if its still not a priority?

**Comment by michaelhaxhiu** [comment]
Technically we need to keep this `daily` as it's a `Bug`, even if it's not as pressing of a priority

**Comment by BhuvaneshPatil** [comment]
Hello Everyone, What is status on this one. Previously ir was happening for some tasks only now it's happening on all tasks.

**Comment by michaelhaxhiu** [comment]
Thanks for the bump @BhuvaneshPatil. cc @marcochavezf any progress here?

**Comment by marcochavezf** [comment]
No update. Catching up after being OOO and still focused on TU which will launch in a few days

**Comment by michaelhaxhiu** [comment]
Still giving @marcochavezf some time to focus on TU. Maybe he can give an update EOW 🙏

**Comment by marcochavezf** [comment]
Sorry no update here, still focused on wave7

**Comment by marcochavezf** [comment]
No update, still working on wave7

**Comment by sophiepintoraetz** [comment]
@marcochavezf - while wave 7 is a priority, it is not at the expense of other chores. This bug has been open for over a month now so can you please make some time (even if it is 30 minutes a day) to move this forward?

**Comment by marcochavezf** [comment]
Took a look quickly at the issue and we're getting an empty `reportSummary` [here](https://github.com/Expensify/Web-Expensify/blob/6b5df2e7be5ef7c027a5ccbf19382bc8329b7196/lib/ReportAPI.php#L4923). Seems the problem is coming from Auth.

**Comment by marcochavezf** [comment]
No update, I expect to continue with this one after fixing another bug with higher priority

**Comment by michaelhaxhiu** [comment]
I'm re-assigning this to another BZ as part of my preparation for Sabbatical (starting Friday).

**Next steps:**

- @marcochavezf is going to resume focus on this very soon as he commented above.
- Check on this and ensure it doesn't fall to the wayside 🙏

**Comment by miljakljajic** [comment]
Thank you, Hax! Enjoy your time off.

**Comment by marcochavezf** [comment]
No update, focused on wave7 and traveling back home

**Comment by marcochavezf** [comment]
I'm swamped with other high-priority items, I will come back to this one after I finish the other pending items

**Comment by marcochavezf** [comment]
No update, but I am prioritizing this one for this week

**Comment by marcochavezf** [comment]
No update

**Comment by marcochavezf** [comment]
No update, focused on TU launch

**Comment by marcochavezf** [comment]
No update

**Comment by marcochavezf** [comment]
No update

**Comment by marcochavezf** [comment]
No update

**Comment by marcochavezf** [comment]
No update

**Comment by marcochavezf** [comment]
No update

**Comment by marcochavezf** [comment]
No update

**Comment by miljakljajic** [comment]
Given this has been internal since August without much movement, is this low priority enough to close?

**Comment by marcochavezf** [comment]
I have a backend fix for this in this [PR](https://github.com/Expensify/Auth/pull/9180). I noticed we'd still need to limit the anonymous user to mark the task as done (I will fix that in a second PR).

**Comment by miljakljajic** [comment]
Offers extended to @hoangzinh and @abdulrahuman5196! I'll pay as soon as you accept them.

**Comment by miljakljajic** [comment]
Both payments complete!
